R/extractFDAFeaturesMethods.R

```diff
@@ -106,77 +124,310 @@ extractFDAFourier = function(trafo.coeff = "phase") {
 #' @export
 #' @family fda_featextractor
 extractFDAWavelets = function(filter = "la8", boundary = "periodic") {
-  assertCharacter(filter)
-  assertChoice(boundary, c("periodic", "reflection"))
 
-  lrn = function(data, target = NULL, col, filter, boundary) {
+  # All possible values for the filters
+  filter.vals = c(
+    paste0("d", c(2, 4, 6, 8, 10, 12, 14, 16, 18, 20)),
+    paste0("la", c(8, 10, 12, 14, 16, 18, 20)),
+    paste0("bl", c(14, 18, 20)),
+    paste0("c", c(6, 12, 18, 24, 30)),
+    "haar"
+  )
+
+  lrn = function(data, target = NULL, col, ...) {
+    assertChoice(filter, filter.vals)
+    assertChoice(boundary, c("periodic", "reflection"))
 
-    requirePackages("wavelets", default.method = "load")
+    vals = learnerArgsToControl(list, ...)
+    return(vals)
+  }
 
-    assertClass(data, "data.frame")
-    assertNumeric(as.matrix(data[, col]))
+  reextract = function(data, target, col, vals, args) {
 
-    df = convertRowsToList(data[, col, drop = FALSE])
-    wtdata = t(dapply(df, fun = function(x) {
-      wt = wavelets::dwt(as.numeric(x), filter = filter, boundary = boundary)
+    requirePackages("wavelets", default.method = "load")
+    data = checkFDCols(data, col)
+
+    # Convert to list of rows and extract wavelets from each time-series.
+    rowlst = convertRowsToList(data)
+    wtdata = t(dapply(rowlst, fun = function(x) {
+      vals$X = as.numeric(x)
+      wt = do.call(wavelets::dwt, vals)
+      # Extract wavelet coefficients W and level scaling coeffictients V
       unlist(c(wt@W, wt@V[[wt@level]]))
     }))
```

Comment:
Having multiple `reextract()` funs inside of various funs is likely to cause trouble as some point. e.g. when it comes to debugging, you do not know anymore which `reextract()` to look for.

Can we outsource them into `helpers_fda.R` and suffix them with some name that they apply to?